### ISSUE TITLE
"smaller" subscription results in original "larger" dataset

### ISSUE THREAD

**Issue by lucnat** [adtribute/pub-sub-lite#4]
When I have a subscription that returns some set of data, then I run it again with different arguments to obtain only a fraction of that data, I will end up with all the original data instead of the smaller dataset. This is different behavior from the vanilla Meteor.subscribe/publish. Is this intentional?

**Comment by npvn** [comment]
Yes, this is an expected behaviour of data loaded via pub-sub-lite. As the package's goal is to minimize wasteful cycles of _loading --> resetting --> reloading_ data, data loaded into Minimongo is currently [not reset](https://github.com/adtribute/pub-sub-lite/issues/3) when Method call's arguments change - it just updates Minimongo with the docs retrieved from the new Method call.

I will think about a mechanism for cases when stricter data control is necessary.

For now, can you work around the issue by having explicit selectors in your queries (e.g. `Docs.find({ argument1, argument2 })` instead of `Docs.find()`, where `argument1` and `argument2` reflect the updated Method call's argument(s))? In fact it's a good practice to always have explicit filter in your client-side queries, even though you may have already filtered the data from the server-side. This practice will help avoid unexpected data when a collection is populated by multiple data sources.
